Src/GoogleApis.Core/Apis/Http/ConfigurableMessageHandler.cs

```diff
@@ -64,22 +74,115 @@ public class ConfigurableMessageHandler : DelegatingHandler
         private readonly IList<IHttpExecuteInterceptor> executeInterceptors =
             new List<IHttpExecuteInterceptor>();
 
-        /// <summary>Gets a list of <see cref="IHttpUnsuccessfulResponseHandler"/>.</summary>
-        public IList<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        /// <summary>
+        /// Gets a copied list of <see cref="IHttpUnsuccessfulResponseHandler"/>.
+        /// <remarks>
+        /// Since version 1.10, the return value had been changed from <c>IList</c> to <c>IEnumerable</c> in order to
+        /// keep this class thread-safe. The returned list is also a copied version of the handlers. More info is
+        /// available on <a href="https://github.com/google/google-api-dotnet-client/issues/592">Issue 592</a>.
+        /// </remarks>
+        /// </summary>
+        public IEnumerable<IHttpUnsuccessfulResponseHandler> UnsuccessfulResponseHandlers
+        {
+            get
+            {
+                lock (unsuccessfulResponseHandlersLock)

```

Comment:
use lock(unsuccessfulResponseHandlersLock.SyncRoot) instead and get rid of the lock objects above? Same below.

Also, this property is a bit contraintuitive in the sense that it returns an iterator over the copy of the original list. I understand that exposing a list here wouldn't be good, because than old code using Add or Remove on this property would be broken silently.
Perhaps  exposing a method GetUnsucessfulResponseHandlers() that returns a snapshot of handlers would be better.

Also, have you considered keeping the old API and marking it as Deprecated?